packages/vue/vue-vanilla/src/complex/ObjectRenderer.vue

```diff
@@ -0,0 +1,96 @@
+<template>
+  <div v-if="control.visible">
+    <dispatch-renderer
+      :visible="control.visible"
+      :enabled="control.enabled"
+      :schema="control.schema"
+      :uischema="detailUiSchema"
+      :path="control.path"
+      :renderers="control.renderers"
+      :cells="control.cells"
+    />
+  </div>
+</template>
+
+<script lang="ts">
+import {
+  JsonFormsRendererRegistryEntry,
+  rankWith,
+  ControlElement,
+  Generate,
+  GroupLayout,
+  UISchemaElement,
+  findUISchema,
+  isObjectControl,
+} from '@jsonforms/core';
+import { defineComponent } from 'vue';
+import {
+  DispatchRenderer,
+  rendererProps,
+  RendererProps,
+  useJsonFormsControlWithDetail,
+} from '../../config/jsonforms';
+import { useNested, useVanillaControl } from '../util';
+import { isEmpty, cloneDeep } from 'lodash';
+
+const controlRenderer = defineComponent({
+  name: 'ObjectRenderer',
+  components: {
+    DispatchRenderer,
+  },
+  props: {
+    ...rendererProps<ControlElement>(),
+  },
+  setup(props: RendererProps<ControlElement>) {
+    const control = useVanillaControl(useJsonFormsControlWithDetail(props));
+    const nested = useNested('object');
+    return {
+      ...control,
+      input: control,
+      nested,
+    };
+  },
+  computed: {
+    detailUiSchema(): UISchemaElement {
+      const uiSchemaGenerator = () => {
+        const uiSchema = Generate.uiSchema(this.control.schema, 'Group');
+        if (isEmpty(this.control.path)) {
+          uiSchema.type = 'VerticalLayout';
+        } else {
+          (uiSchema as GroupLayout).label = this.control.label;
+        }
+        return uiSchema;
+      };
+
+      let result = findUISchema(
+        this.control.uischemas,
+        this.control.schema,
+        this.control.uischema.scope,
+        this.control.path,
+        uiSchemaGenerator,
+        this.control.uischema,
+        this.control.rootSchema
+      );
+
+      if (this.nested.level > 0) {
+        result = cloneDeep(result);
+        result.options = {
+          ...result.options,
+          bare: true,
+          alignLeft:
+            this.nested.level >= 4 || this.nested.parentElement === 'array',
+        };
+      }
```

Comment:
Do you actually use this information in your renderers? This was introduced in the Vue Vuetify renderer set because of its specifics, I don't think we necessarily need the `useNested` in the Vanilla renderers. I would have expected a simple implementation [like in the React Material renderers](https://github.com/eclipsesource/jsonforms/blob/7e0115feeced7711b0768d325566aaa6b054c32b/packages/material-renderers/src/complex/MaterialObjectRenderer.tsx).

Reply:
removed